### ISSUE TITLE
Unable to logout from AngularJS client

### ISSUE THREAD

**Issue by mraible** [stormpath/stormpath-sdk-java#1051]
With AngularJS and its `spLogout` directive, I get the following error when trying to logout from a Spring Boot app. 

```
XMLHttpRequest cannot load http://localhost:8080/logout. The request was redirected to 
'http://localhost:8080/', which is disallowed for cross-origin requests that require preflight.
```

This directive should work since it sends a POST and works with the [cors-app example](https://github.com/stormpath/stormpath-sdk-angularjs/tree/master/example/cors-app). When using the cors-app, an OPTIONS request is followed by a POST request.

![logout-node-options](https://cloud.githubusercontent.com/assets/17892/19505463/d65c5c3c-957f-11e6-91f9-e10c3edd5fdb.png)
![logout-node-post](https://cloud.githubusercontent.com/assets/17892/19505464/d66ab7b4-957f-11e6-9f1b-28cfed44e932.png)

When I try something similar with Spring Boot (and Spring Security), I see OPTIONS, GET and then POST. I have the following configured in Spring to handle cross-domain requests.

``` java
@Bean
public FilterRegistrationBean corsFilter() {
    UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
    CorsConfiguration config = new CorsConfiguration();
    config.setAllowCredentials(true);
    config.addAllowedOrigin("http://localhost:3000");
    config.addAllowedHeader("*");
    config.addAllowedMethod("*");
    source.registerCorsConfiguration("/**", config);
    FilterRegistrationBean bean = new FilterRegistrationBean(new CorsFilter(source));
    bean.setOrder(0);
    return bean;
}
```

I had to add the following in `application.properties` to get the above filter to come before Stormpath.

```
stormpath.web.stormpathFilter.order=1
```

Here's what the network traffic looks like when talking to a Spring Boot backend.

![logout-java-options](https://cloud.githubusercontent.com/assets/17892/19505525/6bbe9dda-9580-11e6-945a-c8eed3a6616f.png)
![logout-java-get](https://cloud.githubusercontent.com/assets/17892/19505523/6bbdde9a-9580-11e6-9e2a-28dadb5a27b3.png)
![logout-java-post](https://cloud.githubusercontent.com/assets/17892/19505524/6bbe81ba-9580-11e6-9e4c-054bbb00a6c7.png)

**Comment by mraible** [comment]
FWIW, this still fails after upgrading to the Stormpath AngularJS SDK 1.1.0. Here's the logs from the Spring Security side:

```
2016-10-19 16:12:51.403  INFO 12593 --- [nio-8080-exec-6] Spring Security Debugger                 :

************************************************************

Request received for POST '/logout':

org.apache.catalina.connector.RequestFacade@1b960148

servletPath:/logout
pathInfo:null
headers:
host: localhost:8080
connection: keep-alive
content-length: 0
accept: application/json
origin: http://localhost:3000
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.143 Safari/537.36
content-type: application/x-www-form-urlencoded
referer: http://localhost:3000/view1
accept-encoding: gzip, deflate
accept-language: en,en-US;q=0.8
cookie: Idea-8e2f32b7=cbf234c8-6a37-4fbf-86a8-6000af9a90a2; access_token=eyJraWQiOiIxWkZRSUNHTjdSWDJZQ08xOFk0NkdORjBUIiwic3R0IjoiYWNjZXNzIiwiYWxnIjoiSFMyNTYifQ.eyJqdGkiOiIxSXp0YmhFSWp0S2Y5YVVUdEhMcXluIiwiaWF0IjoxNDc2OTE1MTY3LCJpc3MiOiJodHRwczovL2FwaS5zdG9ybXBhdGguY29tL3YxL2FwcGxpY2F0aW9ucy8xdTRzb3R1YUllbUQ1TW9MRXVIVFlvIiwic3ViIjoiaHR0cHM6Ly9hcGkuc3Rvcm1wYXRoLmNvbS92MS9hY2NvdW50cy8yNnFJQTNZSFdoeWN4cTNEbUxTU3lBIiwiZXhwIjoxNDc2OTE4NzY3LCJydGkiOiIxSXp0YmR1RHBKMWdXbE9jaDFVYU1qIn0.ROtlVZ6qGdPqVKhq8jGQQmJPasu5zGAPyYvvt0os9-s; refresh_token=eyJraWQiOiIxWkZRSUNHTjdSWDJZQ08xOFk0NkdORjBUIiwic3R0IjoicmVmcmVzaCIsImFsZyI6IkhTMjU2In0.eyJqdGkiOiIxSXp0YmR1RHBKMWdXbE9jaDFVYU1qIiwiaWF0IjoxNDc2OTE1MTY3LCJpc3MiOiJodHRwczovL2FwaS5zdG9ybXBhdGguY29tL3YxL2FwcGxpY2F0aW9ucy8xdTRzb3R1YUllbUQ1TW9MRXVIVFlvIiwic3ViIjoiaHR0cHM6Ly9hcGkuc3Rvcm1wYXRoLmNvbS92MS9hY2NvdW50cy8yNnFJQTNZSFdoeWN4cTNEbUxTU3lBIiwiZXhwIjoxNDgyMDk5MTY3fQ.X1qAJKLgLXVA90iQHS-dXENsmNLv-NeAqC8FXLfy_Uo; JSESSIONID=353B22B62C705DE6348B7FA6234893CB


Security filter chain: [
  WebAsyncManagerIntegrationFilter
  SecurityContextPersistenceFilter
  HeaderWriterFilter
  CsrfFilter
  LogoutFilter
  LoginHandlerFilter
  ContentNegotiationAuthenticationFilter
  RequestCacheAwareFilter
  SecurityContextHolderAwareRequestFilter
  SpringSecurityResolvedAccountFilter
  OAuthAuthenticationSpringSecurityProcessingFilter
  AnonymousAuthenticationFilter
  SessionManagementFilter
  ExceptionTranslationFilter
  FilterSecurityInterceptor
]


************************************************************
```

Note that [the changelog for 1.1.0](https://github.com/stormpath/stormpath-sdk-angularjs/blob/master/CHANGELOG.md#110) says the following:

Logout requests (which makes a POST to `/logout` on our framework integration) are now using `Content-Type: application/x-www-form-urlencoded`, in order to prevent an un-necessary OPTIONS request.

**Comment by mraible** [comment]
I tested this against `master`, which has CORS support from https://github.com/stormpath/stormpath-sdk-java/pull/1001. Logout works, so I think we can mark this one as fixed! Should we change the milestone to `1.2.0` since that's what CORS is targeted at?
